module/os/macos/spl/spl-kmem.c

```diff
@@ -4431,7 +4454,105 @@ spl_free_thread()
 
 		last_spl_free = spl_free;
 
-		new_spl_free = 0LL;
+		new_spl_free = total_memory -
```

Comment:
This is a good place to put in a kstat tunable cap of some sort (per discussions with @lundman ), for example either some maximum number of bytes or some percentage of total_memory.  Remember that total_memory is only half of the actual system memory.   

On real systems this code has run for a bit, both busy and not so busy, kstat.spl.misc.spl_misc.os_mem_alloc has generally not strayed above the minuend even when there is ample memory.  It will fall below that value from time to time, if there is more pressure (userland, UBC-cached other vfses) than arc-filling zfs activity (writes, read-misses).